File: 卡码网/Go/test11.go
//11. 共同祖先
package main

import (
	"fmt"
)

func main(){
	var n int
	for {
		_,err:=fmt.Scan(&n)
		if err!=nil {
			return
		}
		commons:=make([]int,21,21)
		var a,b int
		for i:=0;i<n;i++ {
			fmt.Scan(&a,&b)
			commons[a]=b
		}
		//计算1、2的树高度
		len1,len2:=0,0
		//1、2起点
		start1,start2:=1,2
		for commons[start1]!=0 {
			start1=commons[start1]
			len1++
		}
		for commons[start2]!=0 {
			start2=commons[start2]
			len2++
		}
		if len1>len2 {
			fmt.Println("You are my elder")
		}else if len1<len2 {
			fmt.Println("You are my younger")
		}else{
			fmt.Println("You are my brother")
		}
	}
}

File: 卡码网/Js/test11.js
//11. 共同祖先
const rl=require("readline").createInterface({input:process.stdin});
const iter=rl[Symbol.asyncIterator]();
const readline=async ()=>(await iter.next()).value;

async function main(){
  while(line=await readline()){
    const n=parseInt(line);
    let commons=new Array(21).fill(0);
    for(let i=0;i<n;i++){
      const [a,b]=(await readline()).split(" ").map(Number);
      commons[a]=b;
    }
    let height1=0,height2=0;
    let start1=1,start2=2;
    while(commons[start1]!=0){
      height1++;
      start1=commons[start1];
    }
    while(commons[start2]!=0){
      height2++;
      start2=commons[start2];
    }
    if(height1>height2) console.log("You are my elder");
    else if(height1<height2) console.log("You are my younger");
    else console.log("You are my brother");
  }
}

main();